pkg/provisioner/ironic/ironic.go

```diff
@@ -775,104 +734,133 @@ func (p *ironicProvisioner) UpdateHardwareState() (result provisioner.Result, er
 	return result, nil
 }
 
-func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (updates nodes.UpdateOpts, err error) {
-
-	hwProf, err := hardware.GetProfile(p.host.HardwareProfile())
-
-	if err != nil {
-		return updates, errors.Wrap(err,
-			fmt.Sprintf("Could not start provisioning with bad hardware profile %s",
-				p.host.HardwareProfile()))
-	}
-
-	// image_source
+func (p *ironicProvisioner) getImageUpdateOptsForNode(ironicNode *nodes.Node, imageData *metal3v1alpha1.Image, liveImageData *metal3v1alpha1.LiveImage) (updates nodes.UpdateOpts, err error) {
```

Comment:
Inside this method we have the information we need to derive imageData and liveImageData, so we shouldn't need to pass them as arguments.

Reply:
The reason I did this is because since https://github.com/metal3-io/baremetal-operator/pull/716/commits/2e4e00b2c387d2b35979ecba61f8fb2036bb2fcf from @zaneb the imageData is potentially different depending on where it's called - in `ValidateManagementAccess` it could use the `Status` image on adopt, but when called via setUpForProvisioning->getUpdateOptsForNode we'll only ever want to use the `Spec`?

I wasn't sure if unconditionally doing that check against the status was safe, so I tried to replicate the existing logic, I was worried  about introducing a new corner case in addition to the one @zaneb was fixing..